### PR TITLE
fix(link): [FX-2298] remove fontSize

### PR DESCRIPTION
**Pull request by vshyrokov**
[FX-2298](https://toptal-core.atlassian.net/browse/FX-2298)

### Description

Removed fontSize prop from the Link.

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- `n/a` Annotate all `props` in component with documentation
- `n/a` Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and
1. You have no extra requests.
2. You have optional requests.
   1. Add `nit: ` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because
1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:
1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

### CODE REVIEW

**Review by TomasSlama — APPROVED** [review]
LGTM 👍, should we create alpha package and test it on SP? If it solves all the problems for you?

**Review by OndrejTuma — APPROVED** [review]
Code looks good 👍

**Comment by vshyrokov** [comment]
@TomasSlama it would be great to do that, could you help me with creating an alpha package?

**Comment by TomasSlama** [comment]
@toptal-bot run package:alpha-release

**Comment by TomasSlama** [comment]
@vshyrokov ☝️

**Review by TomasSlama — COMMENTED** [review]
I have checked that Happo tests, and I don't think, that these changes are expected.

File: .changeset/dirty-months-float.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': patch
+---
+
+Removed fontSize prop from the Link
```
Comment (denieler) [inline .changeset/dirty-months-float.md:5]: can we please also explain to users of the Link how to deal if they use fontSize prop right now?

```
BREAKING CHANGE:
...
```

File: .changeset/dirty-months-float.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@toptal/picasso': patch
```
Comment (denieler) [inline .changeset/dirty-months-float.md:2]:
```suggestion
'@toptal/picasso': major
```

File: .changeset/dirty-months-float.md (hunk 1 above)
Comment (vshyrokov) [reply .changeset/dirty-months-float.md:5]: Hey @denieler, sure! I'll update as a major version

**Comment by denieler** [comment]
by visual tests it looks like that all links started to inherit the font size of the container where they are defined, wouldn't this break many and many product pages layouts? do we really want to have such a harsh breaking change?

**Comment by vshyrokov** [comment]
@denieler, as I see, it was working like this before, update was added in the v12 and it seems, it works as expected now.

@TomasSlama, could you confirm?

**Comment by denieler** [comment]
also looks like many unrelated to Link components change their look, for example, `Notification` or `PageBanner`

Screenshot 2021-11-23 at 18 09 22

which shouldn't happen

**Review by denieler — CHANGES_REQUESTED** [review]
other components visual test failing, not related to Link

**Comment by vshyrokov** [comment]
@denieler 

actually, it's correct now

https://app.abstract.com/share/aa04519f-5383-4555-9574-521d7afec32d?collectionLayerId=363d365b-d0db-4db2-a7d3-08c83c092930&mode=design&present=true - fontSize is 13px

same to Body Small link

https://www.figma.com/file/q2nvjiyO2CLqBv4DeJnU3i/Product-Library-Documentation?node-id=218%3A4060

12px, we in UserBadge a Typography 

https://github.com/toptal/picasso/blob/49d75d543244a197e0b0127be8a3e84989f31242/packages/picasso/src/UserBadge/story/Title.example.tsx#L52

**Comment by denieler** [comment]
@vshyrokov can't agree with you:

Font size of the Action (Link) is `14px`
Screenshot 2021-11-23 at 18 39 23

master Picasso - `14px`
Screenshot 2021-11-23 at 18 39 15

**Comment by vshyrokov** [comment]
@denieler we have one more example there
image


https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/0707c36cbb23914c365419f1083c142fcd57f269/files/e1d69403-65c3-4a24-927e-5d5f06eaff22/layers/18C44FCF-A16D-4A25-A48D-5267EA182970?collectionId=c39c2374-aeec-40e3-bfb4-18878dde3a4d&collectionLayerId=363d365b-d0db-4db2-a7d3-08c83c092930&mode=build&selected=root-E5B3DFB3-559A-4047-A5F7-0C8ADB30D5E0

**Review by denieler — APPROVED** [review]
Yep, my bad, thank you @vshyrokov, I've checked all the visual screenshots updated and it looks like all of them are according to BASE (except some I couldn't find in Abstract 🤷‍♂️ )

**Comment by vshyrokov** [comment]
@toptal-bot run package:alpha-release

**Review by TomasSlama — APPROVED** [review]

**Pull request merged**